zerver/webhooks/freshstatus/view.py

```diff
@@ -0,0 +1,163 @@
+**Note:** {message}
+""".strip()
+
+FRESHSTATUS_MESSAGE_TEMPLATE_INCIDENT_NOTE_CREATED = """
+The following note has been added to the incident: **{title}**
+**Note:** {message}
+""".strip()
+
+FRESHSTATUS_MESSAGE_TEMPLATE_SCHEDULED_MAINTENANCE_PLANNED = """
+The following scheduled maintenance has been opened: **{title}**
+**Description:** {description}
+**Scheduled Start Time:** {scheduled_start_time}
+**Scheduled End Time:** {scheduled_end_time}
+**Affected Services:**
+{affected_services}
+""".strip()
+
+FRESHSTATUS_MESSAGE_TEMPLATE_SCHEDULED_MAINTENANCE_CLOSED = """
+The following scheduled maintenance has been closed: **{title}**
+**Note:** {message}
+""".strip()
+
+FRESHSTATUS_MESSAGE_TEMPLATE_SCHEDULED_MAINTENANCE_NOTE_CREATED = """
+The following note has been added to the scheduled maintenance: **{title}**
+**Note:** {message}
+""".strip()
+
+FRESHSTATUS_SERVICES_ROW_TEMPLATE = "* {service_name}\n"
+FRESHSTATUS_SERVICES_OTHERS_ROW_TEMPLATE = "[and {services_number} more service(s)]"
+FRESHSTATUS_SERVICES_LIMIT = 5
+
+
+@webhook_view("Freshstatus")
+@has_request_variables
+def api_freshstatus_webhook(
+    request: HttpRequest,
+    user_profile: UserProfile,
+    payload: Dict[str, Any] = REQ(argument_type="body"),
+) -> HttpResponse:
+
+    try:
+        body = get_body_for_http_request(payload)
+        subject = get_subject_for_http_request(payload)
+    except KeyError:
+        message = MISCONFIGURED_PAYLOAD_ERROR_MESSAGE.format(
+            bot_name=user_profile.full_name,
+            support_email=FromAddress.SUPPORT,
+        ).strip()
+        send_rate_limited_pm_notification_to_bot_owner(user_profile, user_profile.realm, message)
+
+        return json_error(_("Invalid payload"))
+
+    check_send_webhook_message(request, user_profile, subject, body)
+    return json_success()
+
+
+def get_services_content(services_data: List[Dict[str, Any]]) -> str:
+    services_content = ""
+    for service in services_data[:FRESHSTATUS_SERVICES_LIMIT]:
+        services_content += FRESHSTATUS_SERVICES_ROW_TEMPLATE.format(
+            service_name=service.get("service_name")
+        )
+
+    if len(services_data) > FRESHSTATUS_SERVICES_LIMIT:
+        services_content += FRESHSTATUS_SERVICES_OTHERS_ROW_TEMPLATE.format(
+            services_number=len(services_data) - FRESHSTATUS_SERVICES_LIMIT,
+        )
+    return services_content.rstrip()
+
+
+def get_subject_for_http_request(payload: Dict[str, Any]) -> str:
+    event_data = payload["event_data"]
+    if event_data["event_type"] == "INCIDENT_OPEN" and payload["id"] == "1":
+        subject = FRESHSTATUS_TOPIC_TEMPLATE_TEST
+    else:
+        subject = FRESHSTATUS_TOPIC_TEMPLATE.format(title=payload["title"])
+
+    return subject
+
+
+def get_body_for_http_request(payload: Dict[str, Any]) -> str:
+    event_data = payload["event_data"]
+    if event_data["event_type"] == "INCIDENT_OPEN" and payload["id"] == "1":
+        body = get_setup_webhook_message("Freshstatus")
```

Comment:
Can we write this as `return get_setup_webhook_message` and then skip the `else` statement?

That'd make this function a lot cleaner / more readable.

(I think in general you probably want to just return rather than setting `body` or `subject` in these functions and then returning only once at the end).